spec/services/events/post_process_service_spec.rb

```diff
@@ -0,0 +1,273 @@
+        create(:billable_metric, organization:, aggregation_type: 'sum_agg', field_name: 'item_id')
+      end
+
+      let(:event_properties) { { billable_metric.field_name => '12' } }
+
+      before { charge }
+
+      it 'enqueues a job to perform the pay_in_advance aggregation' do
+        expect { process_service.call }.to have_enqueued_job(Fees::CreatePayInAdvanceJob)
+      end
+
+      context 'when charge is invoiceable' do
+        before { charge.update!(invoiceable: true) }
+
+        it 'does not enqueue a job to perform the pay_in_advance aggregation' do
+          expect { process_service.call }.not_to have_enqueued_job(Fees::CreatePayInAdvanceJob)
+        end
+      end
+
+      context 'when multiple charges have the billable metric' do
+        before { create(:standard_charge, :pay_in_advance, plan:, billable_metric:, invoiceable: false) }
+
+        it 'enqueues a job for each charge' do
+          expect { process_service.call }.to have_enqueued_job(Fees::CreatePayInAdvanceJob).twice
+        end
+      end
+    end
+
+    context 'when event matches a pay_in_advance charge that is invoiceable' do
+      let(:charge) { create(:standard_charge, :pay_in_advance, plan:, billable_metric:, invoiceable: true) }
+      let(:billable_metric) do
+        create(:billable_metric, organization:, aggregation_type: 'sum_agg', field_name: 'item_id')
+      end
+
+      let(:event_properties) { { billable_metric.field_name => '12' } }
+
+      before { charge }
+
+      it 'enqueues a job to create the pay_in_advance charge invoice' do
+        expect { process_service.call }.to have_enqueued_job(Invoices::CreatePayInAdvanceChargeJob)
+      end
+
+      context 'when charge is not invoiceable' do
+        before { charge.update!(invoiceable: false) }
+
+        it 'does not enqueue a job to create the pay_in_advance charge invoice' do
+          expect { process_service.call }
+            .not_to have_enqueued_job(Invoices::CreatePayInAdvanceChargeJob)
+        end
+      end
+
+      context 'when multiple charges have the billable metric' do
+        before { create(:standard_charge, :pay_in_advance, plan:, billable_metric:, invoiceable: true) }
+
+        it 'enqueues a job for each charge' do
+          expect { process_service.call }
+            .to have_enqueued_job(Invoices::CreatePayInAdvanceChargeJob).twice
+        end
+      end
+
+      context 'when value for sum_agg is negative' do
+        let(:event_properties) { { billable_metric.field_name => '-5' } }
+
+        it 'enqueues a job' do
+          expect { process_service.call }
+            .to have_enqueued_job(Invoices::CreatePayInAdvanceChargeJob)
+        end
+      end
+
+      context 'when event field name does not batch the BM one' do
+        let(:event_properties) { { 'wrong_field_name' => '-5' } }
+
+        it 'does not enqueue a job' do
+          expect { process_service.call }
+          .not_to have_enqueued_job(Invoices::CreatePayInAdvanceChargeJob)
```

Comment:
Layout/MultilineMethodCallIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.